src/ServiceControl/ExternalIntegrations/FailedMessageArchivedPublisher.cs

```diff
@@ -0,0 +1,33 @@
+namespace ServiceControl.ExternalIntegrations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Contracts.MessageFailures;
+    using Raven.Client;
+
+    class FailedMessageArchivedPublisher : EventPublisher<FailedMessageArchived, FailedMessageArchivedPublisher.DispatchContext>
+    {
+        protected override DispatchContext CreateDispatchRequest(FailedMessageArchived @event)
+        {
+            return new DispatchContext
+            {
+                FailedMessageId = new Guid(@event.FailedMessageId)
+            };
+        }
+
+        protected override Task<IEnumerable<object>> PublishEvents(IEnumerable<DispatchContext> contexts, IAsyncDocumentSession session)
+        {
+            return Task.FromResult(contexts.Select(r => (object)new Contracts.FailedMessagesArchived
+            {
+                FailedMessagesIds = new[] { r.FailedMessageId.ToString() }
+            }));
+        }
+
+        public class DispatchContext
+        {
+            public Guid FailedMessageId { get; set; }
```

Comment:
We can replace this one by `string FailedMessageId`. Both are serialized to string in json so there should be no problem with deserialization and it will save us the conversions in lines 16 and 24.